smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java

```diff
@@ -65,11 +65,17 @@ public KafkaSource(Vertx vertx, Config config, String servers) {
                 .doOnError(t -> LOGGER.error("Unable to read a record from Kafka topic '{}'", topic, t));
 
         if (config.getOptionalValue("retry", Boolean.class).orElse(true)) {
-            Integer max = config.getOptionalValue("retry-attempts", Integer.class).orElse(5);
-            flowable = flowable
+            Integer max = config.getOptionalValue("retry-attempts", Integer.class).orElse(null);
+
+            if (max == null) {
+                // always retry
+                flowable = flowable.retryWhen(attempts -> attempts.flatMap(i -> Flowable.timer(10, TimeUnit.SECONDS)));
```

Comment:
Wondering if we should implement some kind of exponential backoff. WDYT?

Reply:
Exponential backoff sounds good! Do you think it makes sense to backing off up to a max level?

Reply:
Yes, something like 30 seconds max between attempt.